app/src/MainController.cpp

```diff
@@ -0,0 +1,290 @@
+//
+// Created by marko on 2/2/25.
+//
+
+#include <engine/graphics/GraphicsController.hpp>
+#include <engine/graphics/OpenGL.hpp>
+#include <engine/platform/PlatformController.hpp>
+#include <engine/resources/ResourcesController.hpp>
+#include <MainController.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <imgui.h>
+
+#include "spdlog/spdlog.h"
+
+class MainPlatformEventObserver : public engine::platform::PlatformEventObserver {
+    void on_mouse_move(engine::platform::MousePosition position) override;
+    void on_key(engine::platform::Key key) override;
+    bool m_first_flick     = true;
+    bool cursor_visibility = false;
+};
+
+void MainPlatformEventObserver::on_mouse_move(engine::platform::MousePosition position) {
+    if (m_first_flick) {
+        m_first_flick = false;
+        return;
+    }
+    auto platform = engine::core::Controller::get<engine::platform::PlatformController>();
+    auto graphics = engine::core::Controller::get<engine::graphics::GraphicsController>();
+    auto camera   = graphics->camera();
+    if (platform->get_cursor_status())
+        return;
+
+    camera->rotate_camera(position.dx, position.dy);
+}
+
+void MainPlatformEventObserver::on_key(engine::platform::Key key) {
+    auto platform = engine::core::Controller::get<engine::platform::PlatformController>();
+    auto graphics = engine::core::Controller::get<engine::graphics::GraphicsController>();
+    auto camera   = graphics->camera();
+//engine::platform::KeyId::KEY_SPACE).state() == engine::platform::Key::State::JustPressed
+    if (platform->key(engine::platform::KeyId::KEY_F12).is_down()) {
+        cursor_visibility = !cursor_visibility;
+        platform->set_cursor_status(cursor_visibility);
+        platform->set_cursor_visible(cursor_visibility);
+    }else if(platform->key(engine::platform::KeyId::KEY_E).is_down()) {
+        auto mainController = platform->get<MainController>();
+        // 0 turning on
+        // 1 shutting down
+        mainController->moon_event = (mainController->moon_event + 1) % 2;
```

Comment:
Šta mislite o ideji da se ovo enkapsulira u MoonEvent klasu, koja bi imala metode poput turn_on, shut_down i možda još neku dodatnu funkciju ako je potrebna? Na sličan način bi se mogli organizovati i ostali događaji.

Reply:
Rekao bih isto kao u https://github.com/marko11120/matf-rg-project-2024/pull/3#discussion_r1950394175
Te promene će se vremenom nakupiti i onda će ih biti mnogo teže srediti.